ledger/complete/mtrie/node/node.go

```diff
@@ -31,12 +31,11 @@ type Node struct {
 	// the current implementation is designed to operate on a sparsely populated
 	// tree, holding much less than 2^64 registers.
 
-	lChild    *Node           // Left Child
-	rChild    *Node           // Right Child
-	height    int             // height where the Node is at
-	path      ledger.Path     // the storage path (dummy value for interim nodes)
-	payload   *ledger.Payload // the payload this node is storing (leaf nodes only)
-	hashValue hash.Hash       // hash value of node (cached)
+	lChild       *Node     // Left Child
+	rChild       *Node     // Right Child
+	height       int       // height where the Node is at
+	hashValue    hash.Hash // hash value of node (cached)
+	leafNodeHash hash.Hash // hash value of the fully expanded leaf node (leaf nodes only)
```

Comment:
This is the main change for https://github.com/onflow/flow-go/issues/3843

The payload and path on the leaf node will be replaced by the leaf node hash for retrieving the payload from storage.